### ISSUE TITLE
Set global humanization using MIDI and OSC commands

### ISSUE THREAD

**Issue by theGreatWhiteShark** [hydrogen-music/hydrogen#1684]
As requested in #1683 we could add MIDI and OSC commands to alter timing and velocity humanization as well as swing.

**Comment by devmaiter** [comment]
Hi, I'd like to work on this if it's still open

**Comment by devmaiter** [comment]
Hi, I'd like to work on this if it's still open

**Comment by theGreatWhiteShark** [comment]
Hey @devmaiter. Sure! Please target the `main` branch

**Comment by devmaiter** [comment]
Hi @theGreatWhiteShark, thanks! Before starting I went through `main` and it looks like this was already implemented in df52183d ("MidiAction: add humanization actions", Jan 2026) — timing/velocity/swing, absolute and relative, for both MIDI and OSC, with tests. Is there anything still missing here (documentation, OSC feedback, something from #1683?), or can this issue be closed? Happy to pick up whatever remains, or another issue if this one is done.

**Comment by theGreatWhiteShark** [comment]
Ah, shit. And I only closed the original ticket and not this one 😬 . Sorry for the waste of time!

> Is there anything still missing here (documentation, OSC feedback, something from https://github.com/hydrogen-music/hydrogen/discussions/1683?)

The documentation is a huge mess right now. The current one is based on docbook + XSLT (ancient technology). For the 2.0 release I plan to convert our Jekyll-based web page to Docusaurus and integrate the docs in their directly. I already started but it is far from usable right now. But I don't think I have time to bring this to an productive level until an alpha version of the 2.0 release is ready.

I wonder if anybody actually has used OSC feedback. I would wait till someone requests such a feature. 

> can this issue be closed

Yes, everythings done in here. Doc has to wait.

> Happy to pick up whatever remains, or another issue if this one is done.

Are you interested in a specific use case or a specific part of Hydrogen, like UI/UX, new features, or interoperability?

**Comment by theGreatWhiteShark** [comment]
Implemented in https://github.com/hydrogen-music/hydrogen/commit/df52183d7db743fb362eca3f60fe0676cff973b2

**Comment by devmaiter** [comment]
Thanks a lot for the detailed answer, @theGreatWhiteShark — and no worries at all, going through the humanization code was a good way to get familiar with the codebase!

To answer your question: **interoperability** is what interests me most (I enjoyed digging into the MIDI/OSC action code), and I'm also a daily **Windows** user, so I can help reproduce and work on Windows-specific issues — I noticed a few open ones (#1625, #2181, #2170) and that platform seems to have less coverage.

My plan is to start with a `good first issue` (#2068 or #1943) to learn the contribution workflow, and then move into those two areas. If you'd point me somewhere else first, happy to follow your lead. Thanks again!

**Comment by theGreatWhiteShark** [comment]
Hey @devmaiter,

Great to hear. Tackling those Windows issues would be really helpful! 

> My plan is to start with a good first issue (https://github.com/hydrogen-music/hydrogen/issues/2068 or https://github.com/hydrogen-music/hydrogen/issues/1943) to learn the contribution workflow, and then move into those two areas. If you'd point me somewhere else first, happy to follow your lead. Thanks again!

#2068 is mostly an UI issue. If you not into designing SVGs, this won't probably be much fun. Other than that I'm eagerly awaiting you pull requests :)

As a side note: I'm currently working on a ~large~ massive prototype to make Hydrogen a CLAP/LV2/VST3 plugin (see https://github.com/hydrogen-music/hydrogen/tree/feat/plugin). I'm not sure yet whether this will work out yet. But if it lands, it will bring some heavy structural changes to the repo, like removal of all singletons and GUI/core split
